crates/wash-lib/src/build/provider.rs

```diff
@@ -0,0 +1,132 @@
+use std::io::ErrorKind;
+use std::path::PathBuf;
+use std::process;
+
+use anyhow::{anyhow, bail, Context, Result};
+use nkeys::KeyPairType;
+use tracing::warn;
+
+use crate::build::SignConfig;
+use crate::cli::par::{create_provider_archive, detect_arch, ParCreateArgs};
+use crate::cli::{extract_keypair, OutputKind};
+use crate::parser::{CommonConfig, LanguageConfig, ProviderConfig};
+
+/// Build a capability provider for the current machine's architecture
+/// and operating system using provided configuration.
+pub(crate) async fn build_provider(
+    provider_config: &ProviderConfig,
+    language_config: &LanguageConfig,
+    common_config: &CommonConfig,
+    signing_config: Option<&SignConfig>,
+) -> Result<PathBuf> {
+    let LanguageConfig::Rust(rust_config) = language_config else {
+        bail!("Unsupported language for provider: {:?}", language_config)
+    };
+    let mut command = match rust_config.cargo_path.as_ref() {
+        Some(path) => process::Command::new(path),
+        None => process::Command::new("cargo"),
+    };
```

Comment:
```suggestion
    let mut command = process::Command::new(rust_config.cargo_path.as_ref().unwrap_or("cargo"));
```

Reply:
I also endorse this nit

Reply:
Oh we have it like this because `cargo_path` is a PathBuf but "cargo" is a &str. Will see if I can make type checker happy

Reply:
I cannot without making it uglier 😄